### ISSUE TITLE
Where are you exploring?

### ISSUE THREAD

**Issue by tessavdheiden** [shariqiqbal2810/maddpg-pytorch#19]
Hi!

Great repo, I am glad that you implemented the code in Pytorch!

If you use exploration=True, are you really exploring? Epsilon=0.0 always, so you would never use a random selected action, or am I incorrect?

- step called: https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L79
- action computed with exploration=True: https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/utils/agents.py#L55
- action selected, but not with epsilon: https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/utils/agents.py#L67

**Comment by shariqiqbal2810** [comment]
Hi! The exploration is performed inside the `gumbel_softmax` function which obtains differentiable samples from the probability distribution specified by the policy via the [Gumbel-Softmax reparameterization](https://arxiv.org/abs/1611.01144)
